src/utils/bintools.ts

```diff
@@ -284,11 +284,36 @@ export default class BinTools {
   addressToString = (hrp:string, chainid:string, bytes:Buffer)
   :string => `${chainid}-${bech32.encode(hrp, bech32.toWords(bytes))}`;
 
-  stringToAddress = (address:string):Buffer => {
-    const parts:Array<string> = address.trim().split('-');
-    if(parts[1].startsWith("0x") || parts[1].match(/^[0-9A-F]+$/i)){
+  stringToAddress = (address: string, hrp?: string): Buffer => {
+    const parts: string[] = address.trim().split('-');
+
+    if(parts[0].length < 1) {
```

Comment:
`parts[0]` will exist even if `address` is empty or has no `-`, right?

Reply:
(or we could just put this after the next if statement)

Reply:
If you pass in no address, like so

```ts
bintools.stringToAddress()
```

You get this error

```zsh
 error TS2554: Expected 1-2 arguments, but got 0.
 45     let f  = bintools.stringToAddress()
                         ~~~~~~~~~~~~~~~~~

src/utils/bintools.ts:287:22
287   stringToAddress = (address: string, hrp?: string): Buffer => {
                             ~~~~~~~~~~~~~~~
An argument for 'address' was not provided.
```

If you pass in an empty string, like so

```ts
bintools.stringToAddress('')
```

`parts` is `['']` and you get

```zsh
Bech32Error: Error - Valid address must have prefix before -
``` 

If you pass in address w/ no `-`, like so

```ts
bintools.stringToAddress('Xavax194jv0pfujykh5st2pn5hwmgl034728uexuth0t')
```

`parts` is `[ 'Xavax194jv0pfujykh5st2pn5hwmgl034728uexuth0t' ]` and you get 

```zsh
Bech32Error: Error - Valid address should include -
```

Reply:
Resolved by following this advice

> (or we could just put this after the next if statement)